clean clang-3.8 compiler warnings

There are several warnings raised when compiling with clang-3.8

```
make[3]: Entering directory '/opt/gst/master/libva/va/glx'                                               
  CC       va_glx.lo      
  CC       va_glx_impl.lo 
va_glx_impl.c:208:19: warning: passing 'const GLubyte *' (aka 'const unsigned char *') to parameter of ty
pe 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]        
        if(strcmp(strExtension, (const GLubyte *)name) == 0)                                             
                  ^~~~~~~~~~~~                                                                           
/usr/include/string.h:140:32: note: passing argument to parameter '__s1' here                            
extern int strcmp (const char *__s1, const char *__s2)                                                   
                               ^                                                                         
va_glx_impl.c:208:33: warning: passing 'const GLubyte *' (aka 'const unsigned char *') to parameter of ty
pe 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]        
        if(strcmp(strExtension, (const GLubyte *)name) == 0)                                             
                                ^~~~~~~~~~~~~~~~~~~~~                                                    
/usr/include/string.h:140:50: note: passing argument to parameter '__s2' here                            
extern int strcmp (const char *__s1, const char *__s2)                                                   
                                                 ^  
2 warnings generated.     
 ```

 ```
make[3]: Entering directory '/opt/gst/master/libva/va'                                                   
  CC       va.lo                                                                                         
  CC       va_compat.lo   
  CC       va_fool.lo                                                                                    
  CC       va_trace.lo    
  CC       va_tpi.lo      
va_trace.c:343:25: warning: unused variable 'valid' [-Wunused-variable]                                  
    int i = 0, idx = 0, valid = 0;                                                                       
                        ^                                                                                
va_trace.c:738:11: warning: unused variable 'tmp' [-Wunused-variable]                                    
    FILE *tmp;                                                                                           
          ^                                                                                              
va_trace.c:737:20: warning: unused variable 'suffix' [-Wunused-variable]                                 
    unsigned short suffix = 0xffff & ((unsigned int)time(NULL));                                         
                   ^                                                                                     
va_trace.c:826:16: warning: unused variable 'j' [-Wunused-variable]                                      
    int i = 0, j = 0;                                                                                    
               ^                                                                                         
va_trace.c:3566:9: warning: unused variable 'i' [-Wunused-variable]                                      
    int i;                                          
        ^                 
5 warnings generated.                             

 ```

--- a/va/glx/va_glx_impl.c
+++ b/va/glx/va_glx_impl.c
@@ -205,7 +205,7 @@ static int check_extension3(const char *name)
     for(i = 0; i < nbExtensions; i++)
     {
         const GLubyte *strExtension = glGetStringi(GL_EXTENSIONS, i);
-        if(strcmp(strExtension, (const GLubyte *)name) == 0)
+        if(strcmp((const char *) strExtension, name) == 0)
             return 1;
     }
 

--- a/va/va_trace.c
+++ b/va/va_trace.c
@@ -340,7 +340,7 @@ static VAContextID get_ctx_by_buf(
     struct trace_buf_manager *pbuf_mgr = &pva_trace->buf_manager;
     struct trace_buf_info *pbuf_info = pbuf_mgr->pbuf_info[0];
     VAContextID context = VA_INVALID_ID;
-    int i = 0, idx = 0, valid = 0;
+    int i = 0, idx = 0;
 
     LOCK_RESOURCE(pva_trace);
 
@@ -734,8 +734,6 @@ static void refresh_log_file(
 void va_TraceInit(VADisplay dpy)
 {
     char env_value[1024];
-    unsigned short suffix = 0xffff & ((unsigned int)time(NULL));
-    FILE *tmp;
     struct va_trace *pva_trace = calloc(sizeof(struct va_trace), 1);
     struct trace_context *trace_ctx = calloc(sizeof(struct trace_context), 1);
 
@@ -823,7 +821,7 @@ void va_TraceInit(VADisplay dpy)
 void va_TraceEnd(VADisplay dpy)
 {
     struct va_trace *pva_trace = NULL;
-    int i = 0, j = 0;
+    int i = 0;
 
     pva_trace = (struct va_trace *)(((VADisplayContextP)dpy)->vatrace);
     if(!pva_trace)
@@ -3563,7 +3561,6 @@ static void va_TraceVAEncSequenceParameterBufferVP9(
 {
     VAEncSequenceParameterBufferVP9 *p = (VAEncSequenceParameterBufferVP9 *)data;
     DPY2TRACECTX(dpy, context, VA_INVALID_ID);
-    int i;
 
     va_TraceMsg(trace_ctx, "\t--VAEncSequenceParameterBufferVP9\n");
 

